tests/test_cache_api.py

```diff
@@ -131,6 +143,52 @@ def xy(x, y):  # noqa: F811
     tester.expect_new_entries("xy", "xy_squared")
 
 
+def test_entry_delete(preset_flow):
+    tester = CacheTester(preset_flow)
+
+    tester.flow.get("xy_squared")
+    tester.expect_new_entries("x", "y", "xy", "xy_squared")
+
+    (x_entry,) = [
+        entry for entry in tester.flow.cache.get_entries() if entry.entity == "x"
+    ]
+    x_entry.delete()
+    tester.expect_removed_entries("x")
+
+    (xy_entry,) = [
+        entry for entry in tester.flow.cache.get_entries() if entry.entity == "xy"
+    ]
+    xy_entry.delete()
+    tester.expect_removed_entries("xy")
+
+    tester.flow = tester.flow.to_builder().build()
+    tester.flow.get("xy_squared")
+    tester.expect_new_entries("x", "xy")
+
+
+def test_flow_handles_delete_gracefully(builder):
+    builder.assign("a", 1)
+
+    @builder
+    @bn.memoize(False)
+    def b(a):
+        return a + 1
+
+    @builder
+    @bn.memoize(False)
+    def c(b):
+        return b + 1
+
+    flow = builder.build()
+
+    flow.get("b")
+
+    (b_entry,) = [entry for entry in flow.cache.get_entries() if entry.entity == "b"]
+    b_entry.delete()
+
+    flow.get("c")
```

Comment:
Can you add a comment here explaining why this is a useful test? I understand that it's because you want to refresh cache accessors so that Bionic doesn't throw invalid cache state exception but I'm sure it would be hard to remember later. I initially got confused thinking the above test is already handling this until I saw that the entities for this test aren't memoized.

Reply:
Yeah, great idea.